dragonchain/broadcast_processor/broadcast_processor.py

```diff
@@ -110,6 +120,74 @@ def make_broadcast_futures(session: aiohttp.ClientSession, block_id: str, level:
     return broadcasts
 
 
+def get_level_from_storage_location(storage_location: str):
+    result = re.search("BLOCK/.*?-l([2-5])-", storage_location)
+    if result is None:
+        return None
+    return result.group(1)
+
+
+def get_notification_urls(key: str) -> set:
+    try:
+        urls = set(VERIFICATION_NOTIFICATION[key])
+    except KeyError:
+        urls = set()
+    return urls
+
+
+def sign(message: bytes) -> str:
+    return keys.get_my_keys().make_signature(message, crypto.SupportedHashes.sha256)
+
+
+async def process_verification_notifications(session: aiohttp.ClientSession) -> None:
+    """Main function for the verification notification broadcast system
+
+    Retrieves verifications that need to be processed and sends http requests
+
+    Args:
+        session: aiohttp session for http requests
+    """
+    if VERIFICATION_NOTIFICATION:
+        futures = set()
+        for storage_location in await broadcast_functions.get_notification_verifications_for_broadcast_async():
+            verification_bytes = storage.get(storage_location)
+            level = get_level_from_storage_location(storage_location)
+            signature = sign(verification_bytes)
+            all_endpoints = get_notification_urls("all").union(get_notification_urls(f"l{level}"))
+            for url in all_endpoints:
+                future = send_notification_verification(session, url, verification_bytes, signature, storage_location)
+                futures.add(asyncio.ensure_future(future))
+    try:
+        await asyncio.gather(*futures)  # "Fire & forget"
+    except Exception:
+        _log.exception("Error while notifying verification")
+
+
+async def send_notification_verification(
+    session: aiohttp.ClientSession, url: str, verification_bytes: bytes, signature: str, redis_list_value: str
+) -> None:
+    """ Send a notification verification to a preconfigured address
+
+    This is the actual async broadcast of a single notification at its most atomic
+
+    Args:
+        session: aiohttp session for http requests
+        url: The url to which bytes should be POSTed
+        verification_bytes: the verification object read from disk as bytes
+        signature: The signature of the bytes, signed by this dragonchain
+        redis_list_value: the key within a redis queue which should be removed after successful http request
+
+    Returns:
+        None
+    """
+    _log.debug(f"Notification -> {url}")
+    resp = await session.post(
+        url=url, data=verification_bytes, headers={"dragonchainId": keys.get_public_id(), "signature": signature}, timeout=HTTP_REQUEST_TIMEOUT
+    )
+    _log.debug(f"Notification <- {resp.status} {url}")
+    await broadcast_functions.remove_notification_verification_for_broadcast_async(redis_list_value)
```

Comment:
removing a specific value from a redis list is O(n) where n is the size of the list. If we're doing this operation for literally every single broadcast, maybe we should consider having this be a set if possible, as removing a member from a set in redis is O(1) ?

Just a thought

Reply:
Good call. Actually swapping this to a set should be pretty simple

Reply:
swapped